### PR TITLE
handle monitor hotplugs and fix a few bugs that it revealed

### PR DESCRIPTION
**Pull request by ikalco**
realized this while working on hyprwm/Hyprland#6875

### CODE REVIEW

**Comment by nonetrix** [comment]
Is this related to comment I left where monitor would go black after turning it on and off unless I moved a window or anything happened on it?

File: src/backend/Session.cpp [hunk 1]
```diff
@@ -303,46 +303,47 @@ void Aquamarine::CSession::dispatchUdevEvents() {
         return;
     }
 
+    dev_t              deviceNum = udev_device_get_devnum(device);
+    SP<CSessionDevice> sessionDevice;
+    for (auto& sDev : sessionDevices) {
+        if (sDev->dev == deviceNum)
+            sessionDevice = sDev;
```
Comment (vaxerski) [inline src/backend/Session.cpp:310]: why is there no break here?

File: src/backend/drm/DRM.cpp [hunk 2]
```diff
@@ -1439,9 +1448,13 @@ bool Aquamarine::CDRMOutput::setCursor(SP<IBuffer> buffer, const Vector2D& hotsp
         return false;
     }
 
-    if (!buffer)
+    if (!buffer) {
+        connector->crtc->pendingCursor.reset();
+        connector->crtc->cursor->front.reset();
```
Comment (vaxerski) [inline src/backend/drm/DRM.cpp:1453]: we can't yank a DRM buffer from KMS while it's being used, will cause glitches.

**Review by vaxerski — APPROVED** [review]
thanks!

**Pull request merged**